src/network.ts

```diff
@@ -53,23 +53,36 @@ export async function launchNetwork() {
     }
     client.on(NetworkClient.Events.TRANSACTION, transactionListener);
 
-    // Subscribe to new addresses (for balance updates and transactions)
     const subscribedAddresses = new Set<string>();
+    const fetchedAddresses = new Set<string>();
+
+    // Subscribe to new addresses (for balance updates and transactions)
+    // Also remove logged out addresses from fetched (so that they get fetched on next login)
     watch(addressStore.addressInfos, () => {
         const newAddresses: string[] = [];
+        const removedAddresses: Set<string> = new Set<string>(subscribedAddresses);
+
         for (const address of Object.keys(addressStore.state.addressInfos)) {
-            if (subscribedAddresses.has(address)) continue;
+            if (subscribedAddresses.has(address)) {
+                removedAddresses.delete(address);
+                continue;
+            }
+
             subscribedAddresses.add(address);
             newAddresses.push(address);
         }
-        if (!newAddresses.length) return;
 
-        console.debug('Subscribing addresses', newAddresses);
-        client.subscribe(newAddresses);
+        if (newAddresses.length > 0) {
+            console.debug('Subscribing addresses', newAddresses);
+            client.subscribe(newAddresses);
+        }
+
+        removedAddresses.forEach((removedAddress) => {
+            fetchedAddresses.delete(removedAddress);
+        });
```

Comment:
I suggest to simply handle the `removedAddresses` before the `return` condition, to not unnecessarily change code that doesn't need changing.

```ts
        removedAddresses.forEach((removedAddress) => {
            fetchedAddresses.delete(removedAddress);
        });

        if (!newAddresses.length) return;

        console.debug('Subscribing addresses', newAddresses);
        client.subscribe(newAddresses);
```